### ISSUE TITLE
[Bug]: Variable mutation does not work in an async function

### ISSUE THREAD

**Issue by ramsaptami** [appsmithorg/appsmith#21898]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Variable mutation does not work in an async function, removing async keyword in the sample snippet returns expected value

### Steps To Reproduce

1. Run the following snippet and observe there is no console statement logged
```
export default {
	firstName: '',
	lastName: '',
	
	person1: {
		first: "John",
		last: "Doe"
	},
	
	fullName: function () {
		let fullName = this.firstName + " " + this.lastName
		return fullName
	},

	printName: async function () {
		console.log(this.fullName.call(this.person1));
	}
}
```

### Environment

Production

### Version

Cloud

**Comment by bharath31** [comment]
@Rishabh-Rathod will this be fixed with #19653?

**Comment by Rishabh-Rathod** [comment]
> @Rishabh-Rathod will this be fixed with #19653?

Yes.
